src/core/worker.js

```diff
@@ -514,6 +514,7 @@ class WorkerMessageHandler {
       return Promise.all([
         pdfManager.ensureDoc("documentInfo"),
         pdfManager.ensureCatalog("metadata"),
+        Promise.resolve(pdfManager.pdfDocument.catalog.hasStructTree()),
```

Comment:
Shouldn't we be using `pdfManager.ensureCatalog("hasStructTree")` here, similar to how we fetch the other metadata just above, to make sure that the catalog is loaded at this point?